### DISCUSSION TITLE
Disabling automatic deletion of queues

### DISCUSSION THREAD

**Discussion by sxwebdev** [riverqueue/river#486] [Ideas]
From this documentation I understood that if the queue was paused for more than 24 hours, it will be automatically deleted. But in our case, the queue can be paused for an indefinite period, for example 2 or 3 days, while the service is under maintenance. Is it possible to somehow disable automatic deletion of queues? It is very important that the tasks and the queue itself are not deleted, but continue to accumulate, and after the service is back up and running, all tasks continue to be executed.

**Comment by bgentry** [comment]
Jobs are never deleted unless they have been in a finalized state (completed, discarded, cancelled) for longer than the job cleaner’s configured retention periods: https://riverqueue.com/docs/maintenance-services#cleaner

The “queue” database record is an ephemeral thing that has no impact on the actual job records, it’s really just there for tracking and to have a place to propagate things like the queue pause state. It’s hardcoded to delete queue records that haven’t been used in 24 hours, something we could easily make configurable if needed. But note that records don’t get deleted just because the queue is paused for that long, only if there is no active client configured to use that queue for >24 hours. Even if the queue is paused, the record will stick around so long as a Client is alive and configured to use it.

**Reply by sxwebdev** [reply]
Thank you for the detailed answer. But what if the queue should be paused for more than 24 hours? For example, a week

**Reply by bgentry** [reply]
If you have any running clients during this time and none of them are configured to use that queue, the record will be deleted. Any clients that start up (after its deletion) configured to use it will immediately start working jobs because there’s no record telling them to stay paused.

Are you planning to have clients running during this maintenance window? do you plan to keep the paused queue listed in their config during this time?

The queue can stay paused indefinitely. The only reason its record will get deleted is if no running client has that queue listed in its config for >24 hours. Otherwise the timestamp on the queue record will continue to be updated by those clients even if the queue is paused, and it won’t get deleted.

**Reply by sxwebdev** [reply]
Now I understand, thanks!
